confdgnmi/testtool/gNMI_Interface/04__Subscribe.robot

```diff
@@ -5,90 +5,105 @@ Test Tags        subscribe
+    ...    "Subscribe" request modes.    No further functionality (such as actually
     ...    polling the device) is tested.
-    [Template]    Verify Subscribe
+    [Template]    Subscribe ${mode} to default path with encoding JSON_IETF
     STREAM
     ONCE
     POLL
 
-Subscribe to non-existent "prefix"
-
-Subscribe to non-existent "path"
-
-# message Subscription {
-#   Path path = 1;                    // The data tree path.
-#   SubscriptionMode mode = 2;        // Subscription mode to be used.
-#   uint64 sample_interval = 3;       // ns between samples in SAMPLE mode.
-#   bool suppress_redundant = 4;
-#   uint64 heartbeat_interval = 5;
-# }
-
 Subscribe ONCE with supported "encoding" values
-
-Two Subscriptions in single request - with same "path"
-
-Two Subscriptions in single request - with different "path"
+    [Tags]    sanity
+    [Documentation]    Verify that the device is able to respond correctly for
+    ...    all declared encodings.
+    Given device capabilities
+    and subscription paths    ${GET-PATH}
+    Then subscribe ONCE with supported encodings
+
+Two subscriptions in single request with the same "path"
+    [Documentation]    Verify that the device can handle ONCE subscription
+    ...    with two identical paths.
+    Given subscription paths    ${GET-PATH}    ${GET-PATH}
+    And subscription ONCE with encoding JSON_IETF
+    Then device responds
+
+Two subscriptions in single request with different "path"
+    [Documentation]    Verify that the device can handle ONCE subscription
+    ...    with two different paths.
+    Given subscription paths    ${GET-PATH}    ${SECONDARY-PATH}
+    And subscription ONCE with encoding JSON_IETF
+    Then device responds
+
+Subscribe ONCE sends final message with "sync_response"
+    [Documentation]    When a ONCE subscription is created, the device must
+    ...    respond with a series of responses terminated by an empty
+    ...    response with "sync_response".
+    Given Subscription paths    ${GET-PATH}
+    And Subscription ONCE with encoding JSON_IETF
+    Then Device sends terminated response series and closes the stream
+
+Subscribe POLL sends final message with "sync_response"
+    [Documentation]    When a POLL subscription is created, the device must
+    ...    send an initial set of responses terminated by an empty
+    ...    response with "sync_response".
+    Given Subscription paths    ${GET-PATH}
+    And Subscription POLL with encoding JSON_IETF
+    Then Device sends terminated response series
+
+Subscribe STREAM sends final message with "sync_response"
+    [Documentation]    When a STREAM subscription is created, the device must
+    ...    send an initial set of responses terminated by an empty
+    ...    response with "sync_response".
+    Given Subscription paths    ${GET-PATH}
+    And Subscription POLL with encoding JSON_IETF
+    Then Device sends terminated response series
 
 QOS marked subscriptions
+    [Tags]    unimplemented
```

Comment:
imho adding also `Skip` KW as test "implementation" to silence some robocop lints allows easier resolution of actually failing tests vs non-existing ones... (same for other instances of "unimplemented" further)